src/neo-asm/ASM_Language/linker/Linker.cs

```diff
@@ -0,0 +1,301 @@
+
+                Neo.VM.ScriptBuilder sb = new VM.ScriptBuilder();
+
+                if (inst.valuetext == null)
+                    throw new Exception("error push arg");
+                if (inst.valuetext[0] == '\'' || inst.valuetext[0] == '"' || inst.valuetext[0] == '[')
+                {//string or bytearray
+                    byte[] data = ParseData(inst.valuetext);
+                    sb.EmitPush(data);
+                }
+                else
+                {//number
+                    var num = ParseNumber(inst.valuetext);
+                    sb.EmitPush(num);
+                }
+                code.finalbytes = sb.ToArray();
+
+            }
+            else if (inst.opcode.opcodeVM == VM.OpCode.JMP || inst.opcode.opcodeVM == VM.OpCode.JMPIF || inst.opcode.opcodeVM == VM.OpCode.JMPIFNOT)
+            {
+                //2 bytes data
+
+                code.JMPTarget = inst.valuetext;
+                if (inst.valuetext[0] == '"' || inst.valuetext[0] == '\'')
+                {
+                    code.JMPTarget = inst.valuetext.Substring(1, inst.valuetext.Length - 2);
+                }
+
+                code.CALLTarget = null;
+                code.finalbytes = new byte[3];
+                code.finalbytes[0] = (byte)inst.opcode.opcodeVM;
+            }
+            else if (inst.opcode.opcodeVM == VM.OpCode.CALL)
+            {
+                //2 bytes data
+                code.JMPTarget = null;
+                code.CALLTarget = inst.valuetext;
+                if (inst.valuetext[0] == '"' || inst.valuetext[0] == '\'')
+                {
+                    code.CALLTarget = inst.valuetext.Substring(1, inst.valuetext.Length - 2);
+                }
+                code.finalbytes = new byte[3];
+                code.finalbytes[0] = (byte)inst.opcode.opcodeVM;
+
+            }
+            else if (inst.opcode.opcodeVM == VM.OpCode.SYSCALL)
+            {
+                //4字节参数
+                code.JMPTarget = null;
+                code.CALLTarget = null;
+                code.finalbytes = new byte[5];
+                code.finalbytes[0] = (byte)inst.opcode.opcodeVM;
+
+
+                if (inst.valuetext[0] == '[')
+                {// a data value
+                    var data = ParseData(inst.valuetext);
+                    Array.Copy(data, 0, code.finalbytes, 1, 4);
+                }
+                else if (inst.valuetext[0] == '"')
+                {// a string value
+                    uint api = ToInteropMethodHash(inst.valuetext.Substring(0, inst.valuetext.Length - 2));
+                    var data = BitConverter.GetBytes(api);
+                    Array.Copy(data, 0, code.finalbytes, 1, 4);
+                }
+                else
+                {// a number value
+                    uint api = (uint)ParseNumber(inst.valuetext);
+                    var data = BitConverter.GetBytes(api);
+                    Array.Copy(data, 0, code.finalbytes, 1, 4);
+                }
```

Comment:
else throw exception?

Reply:
if is not a number parseNumber will throw